lib/src/execute.rs

```diff
@@ -355,7 +355,11 @@ impl ExecuteCtx {
         outcome
     }
 
-    pub async fn run_main(self, program_name: &str, args: &[String]) -> Result<(), anyhow::Error> {
+    pub async fn run_main(
+        self,
+        program_name: &str,
+        args: &Vec<String>,
```

Comment:
I would keep this as `&[String]`. It's unusual to see `&Vec` as a function argument since it limits your flexibility on what you can pass. I don't think that flexibility matters all that much here, but it's nice from an unsurprising code perspective